parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -322,7 +328,13 @@ downloadSpinKube(){
         return 
     fi
     
-    retrycmd_if_failure 30 5 60 curl -fSLv -o "$containerd_spinkube_filepath/containerd-shim-spin-v2" "$containerd_spinkube_url/containerd-shim-spin-v2" 2>&1 | tee $CURL_OUTPUT | grep -E "^(curl:.*)|([eE]rr.*)$" && (cat $CURL_OUTPUT && exit $ERR_KRUSTLET_DOWNLOAD_TIMEOUT) &
+    output_file="$containerd_spinkube_filepath/containerd-shim-spin-v2"
+    download_url="$containerd_spinkube_url/containerd-shim-spin-v2"
+    retrycmd_if_failure 30 5 60 curl -fSLv -o "$output_file" "$download_url" 2>&1 | tee $CURL_OUTPUT &
```

Comment:
Similarly with above, We only want the `curl` command to be running in background (with syntax `&`).
Splitting them into multiple steps for easier debugging and can exit with the error code correctly.